index.d.ts

```diff
@@ -0,0 +1,208 @@
+    modified?: string;
+  }
+  interface GetOuput<Schema> extends GetItemOutput {
+    Item?: Schema & SchemaBase;
+  }
+  interface PutOutput<Schema> extends PutItemOutput {
+    Attributes?: Schema & SchemaBase;
+  }
+  interface DeleteOutput<Schema> extends DeleteItemOutput {
+    Attributes?: Schema & SchemaBase;
+  }
+  export class Entity<Schema> {
+    constructor(options: EntityConstructor);
+
+    // Properties
+    table: Table;
+    readonly DocumentClient: DocumentClient;
+    autoExecute: boolean;
+    autoParse: boolean;
+    readonly partitionKey: string;
+    readonly sortKey: string;
+
+    // Methods
+    attribute(attribute: string): string;
+    parse(input, include?);
+    get(
+      item: Schema,
+      options?: {
+        consistent?: boolean;
+        capacity?: "none" | "total" | "indexes";
+        attributes?: [] | {};
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<GetOuput<Schema>>;
+    delete(
+      key: {},
+      options?: {
+        conditions?: [] | {};
+        capacity?: "none" | "total" | "indexes";
+        metrics?: "none" | "size";
+        returnValues?: "none" | "all_old";
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<DeleteOutput<Schema>>;
+    put(
+      item: Schema,
+      options?: {
+        conditions?: [] | {};
+        capacity?: "none" | "total" | "indexes"; // Cannot use DynamoDB type as it's upper-case
+        metrics?: "none" | "size";
+        returnValues?: "none" | "all_old";
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<PutOutput<Schema>>;
+    update(
+      key: Schema & { $remove?: string[]; [string]?: any },
+      options?: {
+        conditions?: [] | {};
+        capacity?: "none" | "total" | "indexes";
+        metrics?: "none" | "size";
+        returnValues?:
+          | "none"
+          | "all_old"
+          | "updated_old"
+          | "all_new"
+          | "updated_new";
+        autoExecute?: boolean;
+        autoParse?: boolean;
+      },
+      parameters?
+    ): Promise<UpdateOutput<Schema>>;
+    query();
+    scan();
+  }
+
+  // Table
+  interface TableAttributes {
+    [attribute: string]: AnyDynamoDBType;
+  }
+  interface Indexes {
+    [index: string]: { partitionKey?: string; sortKey?: string };
+  }
+  interface TableConstructor {
+    name: string;
+    alias?: string;
+    partitionKey: string;
+    sortKey?: string;
+    entityField?: boolean | string;
+    attributes?: TableAttributes;
+    indexes?: Indexes;
+    autoExecute?: boolean;
+    autoParse?: boolean;
+    DocumentClient?: DocumentClient;
+    entities?: {}; // improve - not documented
+  }
+
+  // Entity
+  interface EntityAttributesConfiguration {
+    type?: DynamoDBType;
+    coerce?: boolean;
+    default?: string | boolean | number | [] | {} | ((data: any) => any); // Value "type" or a function
```

Comment:
Would be nice to have 
```suggestion
    default?: string | boolean | number | [] | {} | ((data: Schema) => any); // Value "type" or a function
```
and even some way to constrain the return type to `type`. Maybe at least `(data: Schema) => AnyDynamoDBType`?